### PR TITLE
Adds Doc Source Oracle

### PR DESCRIPTION
**Pull request by marcosmarxm**

### CODE REVIEW

**Review by sherifnada — APPROVED** [review]
awesome. thanks a lot @marcosmarxm !

**Comment by sherifnada** [comment]
will merge once #2697 is merged

**Review by cgardens — COMMENTED** [review]
amazing!

**Pull request merged**
